Sources/TuistGenerator/Settings/DefaultSettingsProvider.swift

```diff
@@ -0,0 +1,115 @@
+        "CLANG_ANALYZER_NONNULL",
+        "CLANG_ANALYZER_NUMBER_OBJECT_CONVERSION",
+        "CLANG_CXX_LANGUAGE_STANDARD",
+        "CLANG_CXX_LIBRARY",
+        "CLANG_ENABLE_MODULES",
+        "CLANG_ENABLE_OBJC_ARC",
+        "CLANG_ENABLE_OBJC_WEAK",
+        "COPY_PHASE_STRIP",
+        "ENABLE_STRICT_OBJC_MSGSEND",
+        "GCC_C_LANGUAGE_STANDARD",
+        "GCC_NO_COMMON_BLOCKS",
+        "PRODUCT_NAME",
+    ]
+
+    private static let essentialTargetSettings: Set<String> = [
+        "SDKROOT",
+        "CODE_SIGN_IDENTITY",
+        "LD_RUNPATH_SEARCH_PATHS",
+        "VALIDATE_PRODUCT",
+        "SWIFT_OPTIMIZATION_LEVEL",
+        "SWIFT_ACTIVE_COMPILATION_CONDITIONS",
+        "CURRENT_PROJECT_VERSION",
+        "DEFINES_MODULE",
+        "DYLIB_COMPATIBILITY_VERSION",
+        "DYLIB_CURRENT_VERSION",
+        "DYLIB_INSTALL_NAME_BASE",
+        "INSTALL_PATH",
+        "PRODUCT_NAME",
+        "SKIP_INSTALL",
+        "VERSION_INFO_PREFIX",
+        "VERSIONING_SYSTEM",
+        "TARGETED_DEVICE_FAMILY",
+        "EXECUTABLE_PREFIX",
+        "COMBINE_HIDPI_IMAGES",
+        "ALWAYS_EMBED_SWIFT_STANDARD_LIBRARIES",
+        "WRAPPER_EXTENSION",
+    ]
+
+    private let settingsHelper = SettingsHelper()
+
+    public init() {}
+
+    // MARK: - DefaultSettingsProviding
+
+    public func defaultProjectSettings(project: Project,
+                                       buildConfiguration: BuildConfiguration) -> [String: Any] {
+        let defaultSettings = project.settings.defaultSettings
+        let variant = settingsHelper.variant(buildConfiguration)
+        let projectDefaultAll = BuildSettingsProvider.projectDefault(variant: .all)
+        let projectDefaultVariant = BuildSettingsProvider.projectDefault(variant: variant)
+        let filter = createFilter(defaultSettings: defaultSettings,
+                                  essentialKeys: DefaultSettingsProvider.essentialProjectSettings)
+
+        var settings: [String: Any] = [:]
+        settingsHelper.extend(buildSettings: &settings, with: projectDefaultAll)
+        settingsHelper.extend(buildSettings: &settings, with: projectDefaultVariant)
+        return settings.filter(filter)
+    }
+
+    public func defaultTargetSettings(target: Target,
+                                      buildConfiguration: BuildConfiguration) -> [String: Any] {
+        let defaultSettings = target.settings?.defaultSettings ?? .recommended
+        let product = settingsHelper.settingsProviderProduct(target)
+        let platform = settingsHelper.settingsProviderPlatform(target)
+        let variant = settingsHelper.variant(buildConfiguration)
+        let targetDefaultAll = BuildSettingsProvider.targetDefault(variant: .all,
+                                                                   platform: platform,
+                                                                   product: product,
+                                                                   swift: true)
```

Comment:
is it always true the targets are swift?

Reply:
Not always, good point, totally forgot to cover "swift false" case.

Reply:
Technically we could omit some of the default swift settings if we pass `false` here. However, currently on master we always pass `true` so the change is compatible with the latest release. I will check if those settings do any harm to Objective-C / C++ only projects. In that case we could probably check the source files to drive it.

Reply:
I'd keep it as it's in master. That's not true for Objective-C but we shouldn't optimize for Objective-C in my opinion. If Objective-C projects want to have a different value they can pass it.

Reply:
Sounds like a sensible idea!